libr/esil/esil_trace.c

```diff
@@ -134,23 +149,41 @@ static bool trace_hook_reg_read(REsil *esil, const char *name, ut64 *res, int *s
 		ret = esil->cb.reg_read (esil, name, res, size);
 	}
 	if (ret) {
-		ut64 val = *res;
+		REsilTraceAccess *access = RVecAccess_emplace_back (&esil->trace->db.accesses);
+		if (!access) {
+			R_LOG_ERROR ("Failed to allocate memory for storing access");
+			return false;
+		}
+
+		access->is_reg = true;
 		// eprintf ("[ESIL] REG READ %s 0x%08"PFMT64x"\n", name, val);
-		sdb_array_add (DB, KEY ("reg.read"), name, 0); // XXX this is very slow and memory hungry
-		sdb_num_set (DB, KEYREG ("reg.read", name), val, 0);
+		access->reg.reg = name;
+		access->reg.value = *res;
+		// TODO size
+		access->is_write = false;
```

Comment:
should we set `end` here to UT64_MAX or 0 or something? otherwise it has uninitialized memory until the next access is set?

Reply:
sounds like a plan

Reply:
Checked it, we implemented it fine the first time. :)